src/dswx_sar/pre_processing.py

```diff
@@ -0,0 +1,400 @@
+import os
+import pathlib
+import time
+import glob
+import numpy as np 
+import logging
+import h5py
+import mimetypes
+
+from osgeo import osr, gdal
+from pathlib import Path
+
+from dswx_sar import filter_SAR 
+from dswx_sar import dswx_sar_util
+from dswx_sar.dswx_runconfig import _get_parser, RunConfig
+
+logger = logging.getLogger('dswx_s1')
+
+def pol_ratio(array1, array2):
+    '''
+    Compute polarimetric coherence from two co-pol and one cross-pol
+    '''
+    array1 = np.asarray(array1, dtype='float32')
+    array2 = np.asarray(array2, dtype='float32')
+
+    return array1 / array2
+
+def pol_coherence(array1, array2, array3):
+    '''Compute polarimetric coherence from two co-pol and one cross-pol
+    '''
+    array1 = np.asarray(array1, dtype='float32') # VVVV
+    array2 = np.asarray(array2, dtype='float32') # VHVH
+    array3 = np.asarray(array3, dtype='complex') # VVVH
+
+    return np.abs(array3 / np.sqrt(array1 * array2))
+
+class AncillaryRelocation:
+
```

Comment:
I think adding a docstring for the class `AncillaryRelocation` would be helpful to understand